code/clue.js

```diff
@@ -1,59 +1,267 @@
+    mystery.favouriteWeapon = randomSelector(weapons)
+  }
+ 
+// This calls/executes the above function.
+
+  shuffleFavouriteWeapon();
+
+// These connects the js and the html, and outputs in browser.
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
+  document.getElementById('favouriteWeapon').innerHTML = `Favourite weapon: ${mystery.favouriteWeapon.name}`
   document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}` 
+  document.getElementById('killerImage').src = mystery.killer.image
+  document.getElementById('killerAge').innerHTML = `${mystery.killer.age}` 
+  document.getElementById('killerOccupation').innerHTML = `${mystery.killer.occupation}` 
+  document.getElementById('killerDescription').innerHTML = `${mystery.killer.description}` 
+};
+
+// FUNCTION - this function will runs when you click on the weapon card.
+
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
+  document.getElementById('loader2').style.display = 'block';
+
+  setTimeout(afterLoaderPickWeapon, 1600);
+};
+
+document.getElementById('loader2').style.display = 'none';
+document.getElementById('weaponCard').addEventListener('click', pickWeapon);
+
+// FUNCTION - this function runs when loader is done.
+
+const afterLoaderPickWeapon = () => {
+  document.getElementById('loader2').style.display = 'none';
+  document.getElementById('weaponCard').removeEventListener('click', pickWeapon);
+  document.getElementById('weaponName').innerHTML = `${mystery.weapon.name}`
+};
+
+// FUNCTION - this function runs when you click on the room card.
+
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms)
+  document.getElementById('loader3').style.display = 'block';
+
+  setTimeout(afterLoaderPickRoom, 1600);
+};
+
+document.getElementById('loader3').style.display = 'none';
+document.getElementById('roomCard').addEventListener('click', pickRoom);
+
+// FUNCTION - This function runs when loader is done.
+
+const afterLoaderPickRoom = () => {
+  document.getElementById('loader3').style.display = 'none';
+  document.getElementById('roomCard').removeEventListener('click', pickRoom);
+  document.getElementById('roomName').innerHTML = `${mystery.room}` 
+};
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+// Creates a function - revealMystery - that runs when you click reveal button.
 
-// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
-// 'The murder was committed by Jacob Green, in the living room with a rope.'
+const revealMystery = () => {
```

Comment:
I like how you used the undefined part of your code! Great!

Reply:
Agreed, and good usage of strict equality (`===`)